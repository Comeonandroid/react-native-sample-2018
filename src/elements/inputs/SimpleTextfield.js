import React, { PureComponent } from 'react'
import { TextInput } from 'react-native'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'
import { prop } from 'styled-tools'

class SimpleTextfield extends PureComponent {

  constructor(props) {
    super(props)

    this.handleTextChange = this.handleTextChange.bind(this)
  }

  handleTextChange(value) {
    const { onChange } = this.props
    if (onChange) {
      onChange({
        value,
        name: this.props.name
      })
    }
  }

  render() {
    const { name, value, ...rest } = this.props
    return (
      <Input
        {...rest}
        name={name}
        value={value}
        onChange={this.handleTextChange}
      />
    )
  }

}

SimpleTextfield.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func
}

const Input = styled(TextInput) `
  border: 1px solid ${prop('theme.colors.inputLightBorder')}
  padding: 2px 5px;
  flex: 1;
  max-width: 120px;
  font-size: 11px;
`

export default SimpleTextfield