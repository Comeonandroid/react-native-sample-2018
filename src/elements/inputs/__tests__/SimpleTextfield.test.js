import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { itBehavesLike } from '../../../specs/shared_examples/index'

import SimpleTextfield from '../SimpleTextfield'

const setup = propOverrides => {
  const props = Object.assign({
    name: 'name',
    value: 'value',
    children: null,
    onChange: jest.fn()
  }, propOverrides)

  const component = <SimpleTextfield {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<SimpleTextfield />', () => {
  const { props, component, wrapper } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  it('properly fires `onChange` when input changes', () => {
    const value = 'Test'
    wrapper.simulate('change', value)
    expect(props.onChange).toHaveBeenCalledWith({ name: props.name, value })
  })
})
