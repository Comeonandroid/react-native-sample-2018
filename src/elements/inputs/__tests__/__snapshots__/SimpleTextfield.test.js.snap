// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleTextfield /> dark: renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  name="name"
  onChange={[Function]}
  style={
    Array [
      Object {
        "borderColor": "#464242",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "fontSize": 11,
        "maxWidth": 120,
        "paddingBottom": 2,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 2,
      },
      undefined,
    ]
  }
  value="value"
/>
`;

exports[`<SimpleTextfield /> light: renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  name="name"
  onChange={[Function]}
  style={
    Array [
      Object {
        "borderColor": "#c1c1c1",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "fontSize": 11,
        "maxWidth": 120,
        "paddingBottom": 2,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 2,
      },
      undefined,
    ]
  }
  value="value"
/>
`;
